shell/platform/linux/fl_texture_registrar_test.cc

```diff
@@ -104,3 +114,22 @@ TEST(FlTextureRegistrarTest, MarkTextureFrameAvailable) {
   EXPECT_TRUE(
       fl_texture_registrar_mark_texture_frame_available(registrar, texture));
 }
+
+// Test the textures can be accessed via multiple threads without
+// synchronization issues.
+TEST(FlTextureRegistrarTest, RegistrarRegisterTextureInMultipleThreads) {
+  g_autoptr(FlEngine) engine = make_mock_engine();
+  g_autoptr(FlTextureRegistrar) registrar = fl_texture_registrar_new(engine);
+  pthread_t threads[kThreadCount];
+
+  for (uint64_t t = 0; t < kThreadCount; t++) {
+    EXPECT_EQ(pthread_create(&threads[t], NULL, add_mock_texture_to_registrar,
+                             (void*)registrar),
+              0);
+  }
+  for (uint64_t t = 0; t < kThreadCount; t++) {
+    pthread_join(threads[t], NULL);
+  };
+  EXPECT_EQ(fl_texture_registrar_get_textures_table_size(registrar),
```

Comment:
You could just use `fl_texture_registrar_lookup_texture` to look up each texture - they will be numbered 1-N. That will avoid adding `fl_texture_registrar_get_textures_table_size`.

Reply:
Quite a good way! I will use it later.

Reply:
I removed `fl_texture_registrar_get_textures_table_size` and use `fl_texture_registrar_lookup_texture` to check each texture now. :)